discovery-infra/test_infra/controllers/node_controllers/node_controller.py

```diff
@@ -116,3 +116,27 @@ def set_ram_kib(self, node_name: str, ram_kib: int) -> None:
     @abstractmethod
     def get_machine_cidr(self) -> str:
         pass
+
+    @abstractmethod
+    def attach_interface(self, node_name, network_xml: str) -> Tuple[Any, Any]:
+        pass
+
+    @abstractmethod
+    def add_interface(self, node_name, network_name, target_interface: str) -> str:
+        pass
+
+    @abstractmethod
+    def undefine_interface(self, node_name: str, mac: str):
+        pass
+
+    @abstractmethod
+    def create_network(self, network_xml: str) -> str:
+        pass
+
+    @abstractmethod
+    def get_network_by_name(self, network_name: str) -> str:
```

Comment:
So what's `get_cluster_network`? I find it a little bit confusing.

Reply:
Moreover, from the signature of the function it sound like it's implementation should be:
```python
def get_network_by_name(self, network_name: str) -> str:
    return network_name
```
Because it gets a network name and it returns the network. Or am I missing something?

Reply:
Your right. will fix

Reply:
we first get to node:
```
def get_network_by_name(self, network_name):
        return self.node_controller.get_network_by_name(network_name)
```
which call to node_controller, which pass it to libvirt controller. 
So the node will return the value

Reply:
In case it returns a network libvirt object, then you should change the returned type from `str`.

Reply:
fixed